### DISCUSSION TITLE
A2DP input to I2S output (connected to second ESP32 like a microphone)

### DISCUSSION THREAD

**Discussion by softhack007** [pschatzmann/ESP32-A2DP#225] [Q&A] [answered]
Hi,

would it be possible (with your lib) to use the I2S output like an I2S microphone, i.e. connecting it as input to another ESP32?
I'm thinking of connecting it to an ESP32 that runs "sound reactive WLED", which supports microphone input via I2S.

cheers,
 Frank.

**Comment by pschatzmann** [comment]
I am not sure if I understand you correctly. Are you trying to do something like this:
- https://github.com/pschatzmann/arduino-audio-tools/tree/main/examples/examples-stream/streams-i2s-a2dp
- https://github.com/pschatzmann/arduino-audio-tools/tree/main/examples/examples-basic-api/base-i2s-a2dp

I suggest that you have a closer look at my [Ardiuno-Audio-Tools](https://github.com/pschatzmann/arduino-audio-tools) project

**Reply by softhack007** [reply]
No, what I am trying is exactly the opposite:
A2DP input via Bluetooth -> ESP32 -> I2S 
then connecting I2S (as slave) to another ESP32 that uses the I2S stream like a digital microphone.

I'll check your audio-tools project, too.

**Comment by pschatzmann** [comment, marked as answer]
I never tried to connect two ESP32 via I2S. Others seem to have problems with it: see https://github.com/pschatzmann/arduino-audio-tools/discussions/149
